pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -1187,13 +1187,15 @@ func (b *MasqueradeBindMechanism) createNatRulesUsingNftables(proto iptables.Pro
 			return err
 		}
 
-		err = b.handler.NftablesAppendRule(proto, "nat", "KUBEVIRT_PREINBOUND",
-			strings.ToLower(port.Protocol),
-			"dport",
-			strconv.Itoa(int(port.Port)),
-			"counter", "dnat", "to", b.getVifIpByProtocol(proto))
-		if err != nil {
-			return err
+		if !hasIstioSidecarInjectionEnabled(b.vmi) {
+			err = b.handler.NftablesAppendRule(proto, "nat", "KUBEVIRT_PREINBOUND",
+				strings.ToLower(port.Protocol),
+				"dport",
+				strconv.Itoa(int(port.Port)),
+				"counter", "dnat", "to", b.getVifIpByProtocol(proto))
+			if err != nil {
+				return err
+			}
```

Comment:
Would you export this to a function ? `performDNATOnPreInbound` or something like that ?